src/main/java/com/cloudbees/jenkins/plugins/bitbucket/client/BitbucketCloudApiFactory.java

```diff
@@ -12,7 +13,7 @@
 public class BitbucketCloudApiFactory extends BitbucketApiFactory {
     @Override
     protected boolean isMatch(@Nullable String serverUrl) {
-        return serverUrl == null;
+        return serverUrl == null || BitbucketCloudEndpoint.SERVER_URL.equals(serverUrl);
```

Comment:
Maybe let consumer pass some extra flag somehow to say given server url is cloud vs standalone server? Maybe use an extension point to set cloud server url?